src/lib.rs

```diff
@@ -1882,6 +1882,33 @@ pub trait Itertools : Iterator {
         self.all(move |elt| used.insert(elt))
     }
 
+    /// Check whether all elements compare equal, and return `Some` with the equal element.
+    ///
+    /// Empty iterators return `None`, and so do iterators with non-equal elements.
+    ///
+    /// ```
+    /// use itertools::Itertools;
+    ///
+    /// let data = vec![1u8, 1, 1, 2, 2, 3, 3, 3, 4, 5, 5];
+    /// assert_eq!(data.iter().equal(), None);
+    /// assert_eq!(data[0..3].iter().equal(), Some(&1));
+    /// assert_eq!(data[3..5].iter().equal(), Some(&2));
+    /// assert_eq!(data[5..8].iter().equal(), Some(&3));
+    ///
+    /// let data : Option<usize> = None;
+    /// assert_eq!(data.into_iter().equal(), None);
+    /// ```
+    fn equal(&mut self) -> Option<Self::Item>
+    where
+        Self: Sized,
+        Self::Item: PartialEq,
+    {
+        match self.next() {
+            None => None,
+            Some(a) => self.all(|x| a == x).then(|| a),
```

Comment:
`bool::then` was added in Rust 1.50, and I do not think we want to bump our minimum required Rust version because of this. Maybe you can change this to a simple `if/else`.

Reply:
Thanks for the hint. Addressed in 626b84428a6d50a97e869a7c06eb44a3dc9904f7 .